### PR TITLE
Don't inspect doc comments for SpecialTypes

### PR DESCRIPTION
**Pull request by rchande**
They don't contain the <completionlist> tag in their documentation.

### CODE REVIEW

**Comment by rchande** [comment]
Want to take a look @pharring @heejaechang ?

/cc: @Pilchie @balajikris @basoundr @brettfo @dpoeschl @jasonmalinowski

File: src/Features/CSharp/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs [hunk 1]
```diff
@@ -121,6 +121,13 @@ public override bool IsTriggerCharacter(SourceText text, int characterPosition,
 
         private INamedTypeSymbol GetCompletionListType(ITypeSymbol type, INamedTypeSymbol within, Compilation compilation)
         {
+            // PERF: None of the SpecialTypes include <completionlist> tags,
+            // so we don't even need to load the documentation.
```
Comment (heejaechang) [inline src/Features/CSharp/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs]: nice. 

for others, can this be cached somehow? are we currently load or parse those xml repeatedly on every compilation change?

**Comment by heejaechang** [comment]
:+1:

File: src/Features/CSharp/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs (hunk 1 above)
Comment (rchande) [inline src/Features/CSharp/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs]: It could be worse--we only have to load documentation for the 1 type we're going to preselect and there typically isn't one. It would be pretty easy to have a dictionary of symbol to symbol, though.

**Comment by pharring** [comment]
:+1:

File: src/Features/CSharp/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs (hunk 1 above)
Comment (heejaechang) [inline src/Features/CSharp/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs]: since blocking completion list is one of biggest impact on typing, it would be good if we can squeeze perf from it as much as possible.

**Comment by Pilchie** [comment]
:+1:

File: src/Features/CSharp/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs (hunk 1 above)
Comment (rchande) [inline src/Features/CSharp/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs]: What's the expiration policy for this cache? We have a map of SymbolKey->SymbolKey, which we populate the first time we preselect a type. What if the user then changes the documentation for that type? We would need to reread the documentation but have no way of knowing when we need to.

Comment (heejaechang) [inline src/Features/CSharp/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs]: we can add something if we needed to (agree not part of this change), but it would be a big win if we can make completion set faster since it will be one of thing that affects throughput.
